MD2Korg/CerebralCortex-DockerCompose
# Add Discovery Dashboard

# Add Discovery Dashboard to CC Personal Edition

@twhnat Hi Tim! Sorry for the delay since I became really sick last week. Here's the PR:

## Changes

- Added services `dashboardserver` and `chronix` used by the Discovery Dashboard, their volumes and links.
- Added a directory named `discovery_dashboard`, which contains the data importer for the dashboard.
    - The importer is a Jupyter Notebook.
    - Added a volume for Jupyter so that the `admin` user is able to see it.
    - Added two links from Jupyter to `dashboardserver` and `chronix` respectively, so that the importer can talk to the dashboard services.
- Nginx
    - Edited Nginx configuration: `/mhealthdashboard/` routes to the dashboard
    - Added a link from nginx to `dashboardserver` for the above purpose

## Tests

I'm using Vagrantfile.dev from the personal branch of CerebralCortex-Vagrant, changed the `git clone` statement to clone my own fork of this repo to provision the machine.

Everything seems to work really well. Here's how I've tested:

- Open up Jupyter Notebook, login, click `discovery_dashboard`, open the importer notebook.
- Edit the importer configuration, import (there's a dummy data generator at bottom for convenience)
- (I made this importer to be really generic and flexible, with an example bridge that can be used to query the CC API. I've observed that since not all streams follow the same data descriptor format, and there might be future API changes, this is probably the best way to go.)
- After the importer has finished, go to `localhost:8080/mhealthdashboard/` and the dashboard should be ready.

![dashboard](https://user-images.githubusercontent.com/11586716/45587670-e2a86900-b8d6-11e8-99df-65e79fe1ff3b.gif)


---

Thank you very much for your help!
